test/e2e/cypress/integration/profiles/recovery/code/errors.spec.ts

```diff
@@ -75,75 +117,67 @@ context('Account Recovery Errors', () => {
 
-      it('should cause form errors', () => {
+      it("should cause form errors", () => {
         cy.visit(recovery)
 
         cy.get('button[value="code"]').click()
         cy.get('[data-testid="ui/message/4000002"]').should(
-          'contain.text',
-          'Property email is missing.'
+          "contain.text",
+          "Property email is missing.",
+        )
+        cy.get('[name="method"][value="code"]').should("exist")
+      })
+
+      it("is unable to recover the account if the code is incorrect", () => {
+        const identity = gen.identityWithWebsite()
+        cy.registerApi(identity)
+        cy.visit(recovery)
+        cy.get(appPrefix(app) + "input[name='email']").type(identity.email)
+        cy.get("button[value='code']").click()
+        cy.get('[data-testid="ui/message/1060003"]').should(
+          "have.text",
+          "An email containing a recovery code has been sent to the email address you provided.",
         )
-        cy.get('[name="method"][value="code"]').should('exist')
+        cy.get("input[name='code']").type("01234567") // Invalid code
+        cy.get("button[value='code']").click()
+        cy.get('[data-testid="ui/message/4060006"]').should(
+          "have.text",
+          "The recovery code is invalid or has already been used. Please try again.",
+        )
+        cy.noSession()
       })
 
-      it('should cause non-repeating form errors after submitting empty form twice. see: #2512', () => {
+      it("should cause non-repeating form errors after submitting empty form twice. see: #2512", () => {
         cy.visit(recovery)
         cy.get('button[value="code"]').click()
-        cy.location('pathname').should('eq', '/recovery')
+        cy.location("pathname").should("eq", "/recovery")
 
         cy.get('button[value="code"]').click()
         cy.get('[data-testid="ui/message/4000002"]').should(
-          'contain.text',
-          'Property email is missing.'
+          "contain.text",
+          "Property email is missing.",
         )
-        cy.get('form')
+        cy.get("form")
           .find('[data-testid="ui/message/4000002"]')
-          .should('have.length', 1)
-        cy.get('[name="method"][value="code"]').should('exist')
+          .should("have.length", 1)
+        cy.get('[name="method"][value="code"]').should("exist")
       })
 
-      it('is unable to recover the email address if the code is expired', () => {
-        cy.shortLinkLifespan()
-        const identity = gen.identityWithWebsite()
-        cy.registerApi(identity)
-        cy.recoverApi({ email: identity.email })
-        cy.recoverEmailButExpired({ expect: { email: identity.email } })
-
-        cy.get('[data-testid="ui/message/4060005"]').should(
-          'contain.text',
-          'The recovery flow expired'
-        )
-
-        cy.noSession()
-      })
-
-      it('is unable to recover the account if the code is incorrect', () => {
+      it("invalid remote recovery email template", () => {
```

Comment:
Test `is unable to recover the email address if the code is expired` is now missing - for a reason?

Reply:
Just renamed, to better reflect what we are testing, IMO: https://github.com/ory/kratos/pull/2645/files/b48710507909abefc579dfa05c5789b4c3fc91ff#diff-4156044df819e73198f46caad2012dc54e4d204a92f2417ff94d29f4a4676d61R75